diff --git a/kitty_tests/ssh.py b/kitty_tests/ssh.py
index 0174f7df..a2f8dc42 100644
--- a/kitty_tests/ssh.py
+++ b/kitty_tests/ssh.py
@@ -151,7 +151,7 @@ class SSHKitten(BaseTest):
             for sh in self.all_possible_sh:
                 if sh == 'sh' or 'python' in sh:
                     q = shutil.which(launcher)
-                    if q:
+                    if q and 'zsh' not in q:
                         with self.subTest(sh=sh, launcher=q), tempfile.TemporaryDirectory() as tdir:
                             self.check_bootstrap(sh, tdir, test_script='env; exit 0', SHELL_INTEGRATION_VALUE='', launcher=q)
 
